File: 2025/bak/main.lua
local lib = require("lib")

function main()
    local f = io.open("input.txt", "r")
    if not f then
        print("File not found")
        return
    end

    local data = {}
    for line in f:lines() do
        table.insert(data, line)
    end

    local ok = f:close()
    if not ok then
        print("Could not close file")
        return
    end

    local first_answer = lib.first(data)
    print(string.format("First Answer: %s", first_answer))

    local second_answer = lib.second(data)
    print(string.format("Second Answer: %s", second_answer))
end

main()

File: 2025/bak/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "lib.c"

int main() {
    FILE* f = fopen("input.txt", "r");
    if (f == NULL) {
        printf("Could not open file\n");
        return 1;
    }

    if (fseek(f, 0, SEEK_END) != 0) {
        printf("Could not find end of file\n");
        fclose(f);
        return 1;
    }

    long size = ftell(f);
    if (size < 0) {
        printf("Could getting file size\n");
        fclose(f);
        return 1;
    }

    if (fseek(f, 0, SEEK_SET) != 0) {
        printf("Could not reset cursor\n");
        fclose(f);
        return 1;
    }

    char* buffer = malloc(size + 1);
    if (buffer == NULL) {
        printf("Could not allocate buffer\n");
        fclose(f);
        return 1;
    }

    size_t n = fread(buffer, 1, size, f);
    buffer[n] = '\0';

    const char** data = malloc(size + 1);
    if (data == NULL) {
        printf("Could not allocate data\n");
        free(buffer);
        fclose(f);
        return 1;
    }

    size_t length = 0;
    char* line = strtok(buffer, "\n");
    while (line != NULL) {
        data[length++] = line;
        line = strtok(NULL, "\n");
    }

    uint64_t first_answer = first(data, length);
    printf("First Answer: %lu\n", first_answer);

    uint64_t second_answer = second(data, length);
    printf("Second Answer: %lu\n", second_answer);

    free(data);
    free(buffer);
    fclose(f);
    return 0;
}
